### PR TITLE
For #2874, #5375, #5131 - Trigger more mode changes in SessionControl

### PR DESCRIPTION
**Pull request by ekager**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlComponent.kt [hunk 1]
```diff
@@ -148,38 +150,44 @@ fun Observer<SessionControlAction>.onNext(onboardingAction: OnboardingAction) {
 sealed class SessionControlChange : Change {
     data class Change(val tabs: List<Tab>, val mode: Mode, val collections: List<TabCollection>) :
         SessionControlChange()
+
     data class TabsChange(val tabs: List<Tab>) : SessionControlChange()
-    data class ModeChange(val mode: Mode) : SessionControlChange()
+    data class ModeChange(val mode: Mode, val tabs: List<Tab>? = null) : SessionControlChange()
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlComponent.kt]: This is the only nonformatting change. Pass new list of tabs into Mode Change action

File: app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlComponent.kt [hunk 2]
```diff
@@ -148,38 +150,44 @@ fun Observer<SessionControlAction>.onNext(onboardingAction: OnboardingAction) {
...
-                    if (change.expand) {
-                        newExpandedCollection.add(change.collection.id)
-                    } else {
-                        newExpandedCollection.remove(change.collection.id)
+        val reducer: (SessionControlState, SessionControlChange) -> SessionControlState =
+            { state, change ->
+                when (change) {
+                    is SessionControlChange.CollectionsChange -> state.copy(collections = change.collections)
+                    is SessionControlChange.TabsChange -> state.copy(tabs = change.tabs)
+                    is SessionControlChange.ModeChange -> state.copy(
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlComponent.kt]: This is the only non formatting change, when mode changes pass in new list of tabs

File: app/src/main/java/org/mozilla/fenix/home/HomeScreenViewModel.kt [hunk 3]
```diff
@@ -6,8 +6,10 @@ package org.mozilla.fenix.home
 
 import android.os.Parcelable
 import androidx.lifecycle.ViewModel
+import org.mozilla.fenix.browser.browsingmode.BrowsingMode
 
 class HomeScreenViewModel : ViewModel() {
     var layoutManagerState: Parcelable? = null
     var motionLayoutProgress = 0F
+    var mode: BrowsingMode? = null
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/home/HomeScreenViewModel.kt]: Adding the last mode to the view model is to fix #5131

Comment (NotWoods) [reply app/src/main/java/org/mozilla/fenix/home/HomeScreenViewModel.kt]: If we're putting mode in the view model we should probably switch to using LiveData and put all our mode-related code in the model. [I've been planning to do this](https://github.com/mozilla-mobile/fenix/pull/4603#discussion_r312206027) but haven't gotten around to it yet.

Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/home/HomeScreenViewModel.kt]: We eventually want to convert the entire Home Screen to the new arch as well but I just wanted to get a quick fix in for these issues https://github.com/mozilla-mobile/fenix/issues/4597

**Comment by ekager** [comment]
I'm going to look back into this today to make sure this is the right solution. Please don't merge!

**Pull request closed without merging**